jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/HttpServerTestBase.java

```diff
@@ -1085,6 +1088,59 @@ public void testCloseWhileWriteBlocked() throws Exception
         }
     }
 
+    @Test
+    public void testCloseWhileCompletePending() throws Exception
```

Comment:
This test clearly shows the major difference between `Response.write(last = true)` and `Callback.succeeded()`.

In my mind, it's always been "the last write sends everything to the client to make it figure out the exchange over" and "succeeding the callback tells the server the exchange is over" but apparently that got broken by accident.

Do we have some documentation speaking about this?

Reply:
I think it is obvious in the contract.  I'm not sure why somebody (probably me) moved the shutdown to the wrong place.... must have been brain fade.